pkg/agent/cniserver/ipam/ipam_service.go

```diff
@@ -72,6 +73,12 @@ func ExecIPAMAdd(cniArgs *cnipb.CniCmdArgs, ipamType string, resultKey string) (
 		return result, nil
 	}
 
+	// Only allocate IP when handling CNI request from infra container.
+	// On windows platform, CNI plugin is called for all containers in a Pod.
+	if !util.IsInfraContainer(cniArgs.Netns) {
```

Comment:
Could we do the check in the caller and decide what to there? Want not to add the infra container check everywhere.
We can add another func just to look up the cached IPAM results, or pass a boolean argument like "allocateNew" to ExecIPAMAdd().

Reply:
Agree with you, it would be better to add checks in caller.

I will move these checks in caller and add `GetIPFromCache()` function to help get the cache results.

Done.